### PR TITLE
Add TDD export button

### LINKED ISSUE #3833: Dashboard: Add "Export" button for the TDD view
Use the API introduced here: https://github.com/rilldata/rill/pull/3390

### PR DESCRIPTION
**Pull request by ericpgreen2**
Closes #3833

Future work:
- Headers of the exported data should use measure `label`  not measure `name` (Platform)
- Sort the export as the UI is sorted
- Offer scheduled reports

### CODE REVIEW

File: web-common/src/features/dashboards/time-dimension-details/TDDExportButton.svelte [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  import exportTDD from "./export-tdd";
+
+  export let includeScheduledReport: boolean;
+  export let metricViewName: string;
+
+  let exportMenuOpen = false;
+  let showScheduledReportDialog = false;
+
+  const ctx = getStateManagers();
+  const metricsView = useDashboard(get(ctx.runtime).instanceId, metricViewName);
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/time-dimension-details/TDDExportButton.svelte:22]: Probably should avoid using `get` inside svelte components and directly use the `$` accessor.

File: web-common/src/features/dashboards/time-dimension-details/export-tdd.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          offset: "0",
+          pivotOn: [timeDimension], // spreads the time dimension across columns
+          priority: undefined,
+          sort: undefined, // future work
+          timeEnd: undefined,
+          timeRange: {
+            start: selectedTimeRange?.start.toISOString(),
+            end: selectedTimeRange?.end.toISOString(),
+          },
+          timeStart: undefined,
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/time-dimension-details/export-tdd.ts:55]: Why do we have two time ranges here?

**Review by djbarnwal — COMMENTED** [review]
App seems to crash on export. Also left a couple of nit comments

image

File: web-common/src/features/dashboards/time-dimension-details/export-tdd.ts (hunk 2 above)
Comment (ericpgreen2) [reply web-common/src/features/dashboards/time-dimension-details/export-tdd.ts]: As discussed, `timeStart` and `timeEnd` are now deprecated – removed from the API call.

**Review by djbarnwal — APPROVED** [review]
Looks good!

**Pull request merged**